### PR TITLE
gitlab: migrate to APIv4

### LINKED ISSUE #583: Gitlab `API V3 is no longer supported. Use API V4 instead.`
```bash
ERROR:bugwarrior.services:Worker for [gitlab] failed: Non-200 status code 410; u'https://gitlab.com/api/v3/projects?per_page=100&page=1'; u'{"error":"API V3 is no longer supported. Use API V4 instead."}'
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/bugwarrior/services/__init__.py", line 506, in _aggregate_issues
    for issue in service.issues():
  File "/usr/local/lib/python2.7/site-packages/bugwarrior/services/gitlab.py", line 402, in issues
    all_repos = self._fetch_paged(tmpl)
  File "/usr/local/lib/python2.7/site-packages/bugwarrior/services/gitlab.py", line 329, in _fetch_paged
    items = self._fetch(tmpl, params=params)
  File "/usr/local/lib/python2.7/site-packages/bugwarrior/services/gitlab.py", line 318, in _fetch
    return self.json_response(response)
  File "/usr/local/lib/python2.7/site-packages/bugwarrior/services/__init__.py", line 485, in json_response
    response.status_code, response.url, response.text,
IOError: Non-200 status code 410; u'https://gitlab.com/api/v3/projects?per_page=100&page=1'; u'{"error":"API V3 is no longer supported. Use API V4 instead."}'
```

### PR DESCRIPTION
**Pull request by chikei**
Also use URL query to exclude tasks other than opened/pending to drastically increase pull speed.

fixes #583

### CODE REVIEW

File: bugwarrior/services/gitlab.py [hunk 1]
```diff
@@ -327,11 +327,11 @@ def filter_repos(self, repo):
         return is_included
 
     def _get_notes(self, rid, issue_type, issueid):
-        tmpl = '{scheme}://{host}/api/v3/projects/%d/%s/%d/notes' % (rid, issue_type, issueid)
+        tmpl = '{scheme}://{host}/api/v4/projects/%d/%s/%d/notes' % (rid, issue_type, issueid)
         return self._fetch_paged(tmpl)
 
     def annotations(self, repo, url, issue_type, issue, issue_obj):
-        notes = self._get_notes(repo['id'], issue_type, issue['id'])
+        notes = self._get_notes(repo['id'], issue_type, issue['iid'])
```
Comment (ralphbean) [inline bugwarrior/services/gitlab.py]: Typo?

Comment (chikei) [reply bugwarrior/services/gitlab.py]: It's a gitlab API change from v3 to v4. When query something belong to merge request or issue, the API was using a site-wide id (the `id` property) to referring MR/issue, in v4 they change it to project-wide id (the `iid` property)

**Comment by Mte90** [comment]
there will be a new release with this fix?

**Comment by wookietreiber** [comment]
A patch I also have applied locally (virtualenv installation of my [wip/wookietreiber branch](https://github.com/wookietreiber/bugwarrior/tree/wip/wookietreiber) that contains all my proposed PRs and some other stuff) is retrying on connection failure: https://github.com/wookietreiber/bugwarrior/commit/74e0c36dca2f023fa462df26ff6db80bfa0121ec

One of the private GitLab instances I used gives spurious network connection resets. Upping the retries to 3 helped a lot. Don't know if that helps in your case, though.

**Comment by plup** [comment]
Same question than @Mte90. Will there be a release with this fix soon or should I build form source for now ?

**Comment by Mte90** [comment]
Also because the last release is of 1 year ago so maybe there are other fix that we didn't know....

**Comment by carlitux** [comment]
I just submited a PR that fix this https://github.com/ralphbean/bugwarrior/pull/613

**Pull request merged**